### ISSUE TITLE
Multiple updates

### ISSUE THREAD

**Issue by XavRsl** [fathomlabs/open-retractions#12]
Hi !
I haven't found multiple updates for a single publication although these must happen regularly (eg. expression of concern -> retraction).  Have I missed them or are you not doing it this way ?

Thanks !

Xavier

**Comment by blahah** [comment]
That's a great point - I think the way the data were initially created overwrote updates with one another - we will need to replace `.update` with `.updates` and have it be an array of objects. Should be able to get this fixed this evening.

**Comment by XavRsl** [comment]
That would be really great !
Also, I find a lot of updates that don't contain timestamps or doi. Would
there be a way to optimize this ?

Thanks !


Le 8 juin 2017 5:11 PM, "Richard Smith-Unna" <notifications@github.com> a
écrit :

That's a great point - I think the way the data were initially created
overwrote updates with one another - we will need to replace .update with
.updates and have it be an array of objects. Should be able to get this
fixed this evening.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/fathomlabs/open-retractions/issues/12#issuecomment-307133691>,
or mute the thread

**Comment by blahah** [comment]
Yup definitely a way to optimise. Do you have any examples of ones without doi/timestamp?

**Comment by XavRsl** [comment]
Sorry, can't give you any DOI tonight I don't have my laptop with me...
Tomorrow morning ?

Le 8 juin 2017 9:15 PM, "Richard Smith-Unna" <notifications@github.com> a
écrit :

**Comment by blahah** [comment]
sure no worries - I'll dig them out myself :)

**Comment by blahah** [comment]
btw are you using the full data dump or the individual files?

**Comment by XavRsl** [comment]
Individual files. Didn't see the full dump...

Le 8 juin 2017 9:29 PM, "Richard Smith-Unna" <notifications@github.com> a
écrit :

**Comment by XavRsl** [comment]
Here's an example DOI : 10.1016/j.jcjd.2015.09.088

Thanks !

**Comment by blahah** [comment]
@XavRsl I've now updated the format so it looks like this:

```
{
  "retracted": false,
  "timestamp": 1470009600000,
  "updates": [
    {
      "timestamp": 1495733219141,
      "identifier": {
        "doi": "10.1194/jlr.d061564err"
      },
      "type": "correction"
    }
  ],
  "identifier": {
    "doi": "10.1194/jlr.d061564"
  },
  "journal": "Journal of Lipid Research",
  "publisher": "American Society for Biochemistry & Molecular Biology (ASBMB)",
  "title": "ERRATUM"
}
```

**Comment by blahah** [comment]
Note that we now have `.identifier.doi` instead of just `.doi` - this is to account for the planned integration of PMIDs and other identifiers. And `.updates` is an array.

I nuked the whole dataset and replaced it with just crossref data for now - I will add in pubmed data tomorrow. I haven't found anything with more than one update, but I'll double check again tomorrow.

**Comment by XavRsl** [comment]
That's great !!! Thank you very much for your time and effort.
I'll try to integrate this this morning. Could you tell me just how you update the API locally ? I've tried ```node update/daily.js```but got an error.

Xavier

**Comment by blahah** [comment]
@XavRsl if you do `npm install` then `node update/daily.js` it should work - I'm doing so using `node v7.9`

**Comment by XavRsl** [comment]
Works ! Don't know what happened last time...

Thanks
